src/ui/public/share/directives/share_object_url.js

```diff
@@ -69,8 +70,14 @@ app.directive('shareObjectUrl', function (Private, Notifier) {
         });
       };
 
+      // since getUrl() is called within a watcher we cache the unhashing step
+      const unhashStatesInAbsUrl = memoize((absUrl) => {
```

Comment:
Yeah, I'm not so concerned about the performance impact as I am with the idea of storing the same value in sessionStorage 50 times just because.